_dev/src/store/modules/product-feed/actions.ts

```diff
@@ -435,22 +435,35 @@ export default {
     }
     commit(MutationsTypes.SET_SELECTED_PRODUCT_CATEGORIES, getSelectedCtg);
   },
-  async [ActionsTypes.GET_PREVALIDATION_PRODUCTS]({rootState}) {
-    const response = await fetch(`${rootState.app.psxMktgWithGoogleApiUrl}/product-feeds/prevalidation-scan`, {
-      method: 'GET',
-      headers: {
-        'Content-Type': 'application/json',
-        Accept: 'application/json',
-        Authorization: `Bearer ${rootState.accounts.tokenPsAccounts}`,
-      },
-    });
+  async [ActionsTypes.GET_PREVALIDATION_PRODUCTS]({rootState, commit, state}) {
+    try {
+      const {limit} = state.preScanDetail;
+      const offset = ((state.preScanDetail.currentPage - 1) * limit).toString();
+      const lang = state.preScanDetail.langChosen;
+      let query = `?limit=${limit}&offset=${offset}`;
 
-    if (!response.ok) {
-      throw new HttpClientError(response.statusText, response.status);
-    }
-    const json = await response.json();
+      if (lang) {
+        query += `&lang=${lang}`;
+      }
+      const response = await fetch(`${rootState.app.psxMktgWithGoogleApiUrl}/product-feeds/prevalidation-scan/errors${query}`, {
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json',
+          Accept: 'application/json',
+          Authorization: `Bearer ${rootState.accounts.tokenPsAccounts}`,
+        },
+      });
 
-    return json;
+      if (!response.ok) {
+        throw new HttpClientError(response.statusText, response.status);
+      }
+      const json = await response.json();
+
+      commit(MutationsTypes.SET_PRESCAN_TOTAL_PRODUCT, json.totalErrors);
+      commit(MutationsTypes.SET_PRESCAN_PRODUCTS, json.errors);
+    } catch (error) {
+      console.log(error);
+    }
```

Comment:
What happens in case of error? Looks like we won't receive it on Sentry, no error message will be displayed and the interface won't change

Reply:
Comment is still valid